packages/core/src/v2/signature.ts

```diff
@@ -251,9 +252,11 @@ export async function recoverTopology(
         throw new Error('Dynamic signature leaf without address')
       }
 
-      const isValid = await isValidSignature(unrecovered.address, subdigest, unrecovered.signature, provider)
-      if (!isValid) {
-        throw new InvalidSignatureLeafError(unrecovered)
+      if (provider) {
```

Comment:
What is the need for adding this? in general, these end up as footguns, as someone using this function never realizes that the signature is never being validate

Reply:
I think it shouldn't be validated by default. `recoverTopology` is about obtaining the tree, not validating the sub signatures. 
Maybe an additional argument would make this clearer.

Reply:
Fair call. The difference with EIP-1271 validation is that it may be conditional on factors outside the signature but ECDSA is not. The function assumes the current point in time is the correct baseline for validation, which it may not be (and isn't for this case). 

I don't think we want to be passing transactions around for this so how about changing the signature to:
```ts
function recoverTopology(
  unrecovered: UnrecoveredTopology,
  subdigest: string,
  provider: ethers.Provider, // Require this so it's more likely to be used
  performERC1271Validation = true // Add this. Default true
)
```

Reply:
That works for me